### PR TITLE
IRO-1142: Increase timeblock to 60 seconds

### PR DESCRIPTION
**Pull request by yuryironfish**
Increase average mining time of block to 60 seconds
Make average block mining time parameterized

### CODE REVIEW

File: ironfish/src/consensus/consensus.ts [hunk 1]
```diff
@@ -48,4 +48,4 @@ export const MAX_REQUESTED_BLOCKS = 50
  * NOTE: This is not used in target calculation, or IRON_FISH_YEAR_IN_BLOCKS.
  * Ideally, it should be in a future refactor.
  */
-export const TARGET_BLOCK_TIME_MS = 15 * 1000
+export const TARGET_BLOCK_TIME_MS = 40 * 1000
```
Comment (NullSoldier) [inline ironfish/src/consensus/consensus.ts:51]: I thought we were going to make this 60 seconds?

File: ironfish/src/primitives/target.test.ts [hunk 2]
```diff
@@ -4,6 +4,12 @@
 
 import { bigIntToBytes, bytesToBigInt, Target, TargetSerde } from './target'
 
+/**
+ * The logic of method calculation the increase/descreas difficulty
```
Comment (NullSoldier) [inline ironfish/src/primitives/target.test.ts:8]: Typo here `descreas`

File: ironfish/src/primitives/target.test.ts [hunk 3]
```diff
@@ -4,6 +4,12 @@
 
 import { bigIntToBytes, bytesToBigInt, Target, TargetSerde } from './target'
 
+/**
```
Comment (NullSoldier) [inline ironfish/src/primitives/target.test.ts:7]: This comment is on the `Describe` for all target tests, but it talks about target calculation. Who is this intended for exactly? Someone trying to learn the target calculation? Or debugging a test and trying to fix the tests?

File: ironfish/src/primitives/target.ts [hunk 4]
```diff
@@ -201,7 +201,7 @@ export class Target {
 
     // max(1 - (current_block_timestamp - parent_timestamp) / 10, -99)
     const diffInSeconds = (time.getTime() - previousBlockTimestamp.getTime()) / 1000
-    const sign = BigInt(Math.max(1 - Math.floor(diffInSeconds / 10), -99))
+    const sign = BigInt(Math.max(1 - Math.floor(diffInSeconds / 40), -99))
```
Comment (NullSoldier) [inline ironfish/src/primitives/target.ts:204]: This doesn't take in the TARGET_TIME_MS variable, can we parameterize this to use that?

**Comment by NullSoldier** [comment]
This PR needs to be rebased, it's merging in 131 commits into staging.

**Comment by yuryironfish** [comment]
Close.
Open new one: https://github.com/iron-fish/ironfish/pull/379

**Pull request closed without merging**